src/nfa/inclusion.cc

```diff
@@ -65,18 +65,27 @@ bool mata::nfa::algorithms::is_included_antichains(
 
         const StateSet& lhs_bigger = lhs.second;
         const StateSet& rhs_bigger = rhs.second;
-        if (lhs_bigger.size() > rhs_bigger.size()) { // bigger set cannot be subset
-            return false;
-        }
 
         //TODO: Can this be done faster using more heuristics? E.g., compare the last elements first ...
         //TODO: Try BDDs! What about some abstractions?
         return lhs_bigger.IsSubsetOf(rhs_bigger);
     };
 
+
     // initialize
-    WorklistType worklist = { };
-    ProcessedType processed(smaller.num_of_states()); // allocate to the number of states of the smaller nfa
+    ProdStatesType worklist = { };//Pairs (q,S) to be processed. It sometimes gives a huge speed-up when they are kept sorted by the size of S,
+    // so those with smaller popped for processing first.
+    ProcessedType processed(smaller.num_of_states()); // Allocate to the number of states of the smaller nfa.
+    // The pairs of each state are also kept sorted. It allows slightly faster antichain pruning - no need to test inclusion in sets that have less elements.
+
+    //Is |S| < |S'| for the inut pairs (q,S) and (q',S')?
+    auto smaller_set = [](const ProdStateType & a, const ProdStateType & b) { return a.second.size() < b.second.size(); };
+
+    //Inserting the pairs (q,S) into a sequence of pairs in order defined by the size of the sets S.
+    auto insert_to_pairs = [&](ProdStatesType & pairs,const ProdStateType & pair) {
+        auto it = std::lower_bound(pairs.begin(), pairs.end(), pair, smaller_set);
+        pairs.insert(it,pair);
+    };
```

Comment:
This might be costly for `deque`. It might be worth investigating whether it can be done more efficiently.

Reply:
Does not take time in the benchmark we were playing with. Also, google says that deque insert and erase are fast.

Reply:
Inserts are, but the lower bound might do something like iterating forward and back time and again to find the position. Not sure about how it will interact with deque. That is what would need to be investigated.

Reply:
Replaced with `std::vector`, which is faster on a hard example: 4 seconds -> 3.6 seconds. We trade fast insert (in deque) for fast lower bound finding (in vector). We will see how the vector performs in our experiments.